### PR TITLE
copr: implement ord considering collation

### LINKED ISSUE #8986: copr: UTF8 functions should consider collation
Some [UTF8 functions](https://github.com/tikv/tikv/blob/master/components/tidb_query_vec_expr/src/impl_string.rs) in TiKV coprocessor do not properly take collation into consideration. For example, [`instr_utf8`](https://github.com/tikv/tikv/blob/44f189c39460ccec75c3f0a5d0210c82eeddc6fa/components/tidb_query_vec_expr/src/impl_string.rs#L561) now compares two strings by their lowercase representation, which is not the right definition.

Related functions:
- [ ] ord (may need to split into ord_utf8)
- [ ] ~~instr_utf8~~
- [x] find_in_set
- [x] strcmp
- [ ]  ~~Locate2ArgsUTF8~~
- [ ] ~~Locate3ArgsUTF8~~

### PR DESCRIPTION
**Pull request by TszKitLo40**
Signed-off-by: Zijie Lu <wslzj40@gmail.com>

### What problem does this PR solve?

Issue Number: close #8986  

Problem Summary:

### What is changed and how it works?

What's Changed:
implement ord considering collation
### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 
- No release note

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

File: components/tidb_query_expr/src/impl_string.rs [hunk 1]
```diff
@@ -1700,20 +1704,27 @@ mod tests {
...
+            (Some("2"), Collation::Utf8Mb4Bin, Some(50i64)),
+            (Some("23"), Collation::Utf8Mb4Bin, Some(50i64)),
+            (Some("2.3"), Collation::Utf8Mb4Bin, Some(50i64)),
+            (Some(""), Collation::Utf8Mb4Bin, Some(0i64)),
+            (Some("你好"), Collation::Utf8Mb4Bin, Some(14990752i64)),
+            (Some("にほん"), Collation::Utf8Mb4Bin, Some(14909867i64)),
+            (Some("한국"), Collation::Utf8Mb4Bin, Some(15570332i64)),
+            (Some("👍"), Collation::Utf8Mb4Bin, Some(4036989325i64)),
+            (Some("א"), Collation::Utf8Mb4Bin, Some(55184i64)),
+            (Some("2.3"), Collation::Utf8Mb4GeneralCi, Some(50i64)),
```
Comment (andylokandy) [inline components/tidb_query_expr/src/impl_string.rs:1716]: Please add both single byte and wide char cases under Latin1bin collation.

Comment (TszKitLo40) [reply components/tidb_query_expr/src/impl_string.rs:1716]: OK, PTAL

**Review by andylokandy — APPROVED** [review]
/lgtm

**Pull request merged**